File: killrvideo-service-videocatalog/src/main/java/com/killrvideo/service/video/dao/LatestVideoRowMapper.java
package com.killrvideo.service.video.dao;

import com.datastax.oss.driver.api.core.cql.Row;
import com.killrvideo.service.video.dto.LatestVideo;
import org.springframework.stereotype.Component;

import static com.killrvideo.dse.dto.Video.*;
import static com.killrvideo.service.video.dto.LatestVideo.COLUMN_YYYYMMDD;

@Component
public class LatestVideoRowMapper {
    public LatestVideo map(Row row) {
        return new LatestVideo(
                row.getString(COLUMN_YYYYMMDD),
                row.getUuid(COLUMN_USERID),
                row.getUuid(COLUMN_VIDEOID),
                row.getString(COLUMN_NAME),
                row.getString(COLUMN_PREVIEW),
                row.getInstant(COLUMN_ADDED_DATE)
        );
    }
}

File: killrvideo-service-videocatalog/src/main/kotlin/com/killrvideo/service/video/dao/LatestVideoRowMapper.kt
package com.killrvideo.service.video.dao

import com.datastax.oss.driver.api.core.cql.Row
import com.killrvideo.service.video.dto.LatestVideo
import com.killrvideo.service.video.dto.Video
import org.springframework.stereotype.Component

@Component
class LatestVideoRowMapper {
    fun map(row: Row): LatestVideo =
        LatestVideo(
            yyyymmdd = row.getString(LatestVideo.COLUMN_YYYYMMDD),
            addedDate = row.getInstant(Video.COLUMN_ADDED_DATE),
            videoid = row.getUuid(Video.COLUMN_VIDEOID),
            userid = row.getUuid(Video.COLUMN_USERID),
            name = row.getString(Video.COLUMN_NAME),
            previewImageLocation = row.getString(Video.COLUMN_PREVIEW)
        )
}
